### PR TITLE
Updates ORF calling Nextflow files

### PR DESCRIPTION
**Pull request by cgpu**
### Overview

This PR updates the Nextflow specific files for the ORF calling modules.

### Description

The changes implement the following:

- [x] remove some boilerplate code to make the files cleaner, minimal
- [x] renames the custom config name to `nextflow.config` for simplicity
- [x] adds boolean param to enable execution of TransDecoder based on this flag (can be removed)
- [x]  modifies channel name to follow the convention **`ch_`<name of channel>** for readability eg. `ch_fasta`
- [x] adds temporary next process to display example of channel  i/o utilisation
- [x] updates the processes containers to  `gsheynkmanlab/transdecoder:5.5.0`
- [x] encourages use of `snake_case` in variable, process, channel names 

### How can I test this works?

Assuming you have cloned the `Long-Read-Proteogenomics` repo, run the following commands to checkout in the branch `cgpu-updates-orf-calling` that implements the change:

```bash

# Navigate into the repo folder
cd Long-Read-Proteogenomics

# Navigate to the branch that has the code
git checkout cgpu-updates-orf-calling

# Navigate to the folder with the Nextflow-ified standalone module:
cd modules/LR_ORFCalling
```

#### 1. **Positive** Nextflow test (should be completed successfully)

```groovy
 nextflow run lr_orfcalling.nf --fasta https://zenodo.org/record/4278034/files/toy_for_christina.fasta.txt
```

#### 2.  **Negative** Nextflow test (should be fail early)

Neglect providing the required parameter `--fasta`. This should fail early with an informative error message.

```groovy
 nextflow run lr_orfcalling.nf
```

### CODE REVIEW

File: modules/LR_ORFCalling/nextflow.config [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  outdir = './results'
+  trans_decoder = true
+  
+  // Boilerplate options
+  help = false
+}
+
+// Ensure that docker is enab
+docker.enabled = true
+process.container = 'gsheynkmanlab/transdecoder:5.5.0'
```
Comment (cgpu) [inline modules/LR_ORFCalling/nextflow.config:22]:
```Suggestion
```

File: modules/LR_ORFCalling/nextflow.config [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  trans_decoder = true
+  
+  // Boilerplate options
+  help = false
+}
+
+// Ensure that docker is enab
+docker.enabled = true
+
+
```
Comment (cgpu) [inline modules/LR_ORFCalling/nextflow.config:23]:
```Suggestion
```

File: modules/LR_ORFCalling/nextflow.config [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  // Boilerplate options
+  help = false
+}
+
+// Ensure that docker is enab
+docker.enabled = true
+
+// Define the exact container we need to use for the transDecoder process:
+process {
+    // Overwrites
```
Comment (cgpu) [inline modules/LR_ORFCalling/nextflow.config:25]:
```Suggestion
```

File: modules/LR_ORFCalling/lr_orfcalling.nf [hunk 4]
```diff
@@ -57,50 +34,63 @@ if (params.help) {
...
-// main.nf
+log.info "--------------------------------------------------------------------------------"
 
-// get run name and date prefix for counts matrix and multiqc
-run_name = params.run_name ? params.run_name + "_" : ""
-date = new Date().format("MM-dd-yy")
-run_prefix = run_name + date
-outdir = run_name
+def summary = [:]
+if (params.fasta) summary['Fasta'] = "${params.fasta}"
```
Comment (cgpu) [inline modules/LR_ORFCalling/lr_orfcalling.nf:40]:
```Suggestion
if (params.fasta) summary['Fasta'] = "${params.fasta}"
summary['TransDecoder'] = "${params.trans_decoder}"
```

**Review by bj8th — COMMENTED** [review]
nextflow successfully run - positive and negative test behaved as expected

**Pull request merged**